rust/connlib/tunnel/src/lib.rs

```diff
@@ -242,9 +253,9 @@ where
         Ok(ConnectionState {
             node: Node::new(private_key, std::time::Instant::now()),
             write_buf: Box::new([0; MAX_UDP_SIZE]),
-            connection_pool_timeout: sleep_until(std::time::Instant::now()).boxed(),
             sockets: Sockets::new()?,
             stats_timer: tokio::time::interval(Duration::from_secs(60)),
+            timeout: Box::pin(tokio::time::sleep_until(Instant::now().into())),
```

Comment:
This is when the connection is first starting? Why is the timeout set to `now`? Just because it must have some value, and `now` is a value that will be in the past so it won't have a spurious wakeup later?

Reply:
Yes.

Reply:
I guess it would be nicer to set it to `None` and thus correctly handle the (hypothetical) case of `Node::poll_timeout` returning `None`.